server/apps/distributed-app/src/main/java/org/apache/james/CassandraRabbitMQJamesServerMain.java

```diff
@@ -205,6 +206,21 @@ public static GuiceJamesServer createServer(CassandraRabbitMQJamesConfiguration
             .combineWith(chooseJmapModule(configuration));
     }
 
+    private static Module chooseMailQueue(CassandraRabbitMQJamesConfiguration configuration) {
+        switch (configuration.getMailQueueChoice()) {
+            case PULSAR:
+                return new PulsarQueueModule();
+            case RABBITMQ:
+                return Modules.combine(
+                    new RabbitMQModule(),
+                    new RabbitMailQueueRoutesModule(),
+                    new RabbitMQMailQueueModule(),
+                    MailQueueViewChoice.ModuleChooser.choose(configuration.getMailQueueViewChoice()));
```

Comment:
Well `RabbitMQModule` should not be in here actually @Arsnael 
Please check again @thanhbv200585

Reply:
Ok seriously I think maybe a renaming or some refactoring should be in order? It seems very rabbitmq specific...

In RabbitMQModule I can see bindings done from conf rabbitmq.properties file, RabbitMQHealthCheck (???), etc. Are we sure there is no impact? Are those common for the two mail queues (honestly the health check I quite doubt it for example)

It's confusing I think for somebody else if he takes a look at the code and see some rabbitmq naming stuff in a pulsar implementation. I'm not convinced we should merge it like that.

@chibenwa thoughts on this?

Reply:
Thanh reallocated the RabbitMQModule: https://github.com/apache/james-project/pull/1651/commits/1e63d9bcd50cacd5eaa9cb00313e3bdcc85512e4

Reply:
Yes and I don't understand why :) Is it normal to have the helath check for rabbitmq injkected for pulsar for example? Cause RabbitMQModule seems to do that, and it got reaffected in the list of general modules on that commit. My concerns are still standing I believe :)

Reply:
> Is it normal to have the helath check for rabbitmq injkected for pulsar for example?

Currently, task queue and event still rely on RabbitMQ, what is wrong with still having RabbitMQHealthCheck?

Reminding you that we are migrating `RabbitMQ alone` -> `RabbitMQ + Pulsar mixed` -> `Pulsar alone`. We did not have the resources to fully migrate to Pulsar in 1 shot.

And on the mail queue module separation, @chibenwa did a pair with Thanh btw...

Reply:
I clearly missed the point of rabbit and pulsar cohabiting for now, apologies.

Alrght then

Reply:
> I clearly missed the point of rabbit and pulsar cohabiting for now, apologies.

No problem, Rene. You spotted the wrong RabbitMQModule placement btw ^^.